stack.go

```diff
@@ -111,6 +113,22 @@ func (c Call) Format(s fmt.State, verb rune) {
 		buf := [6]byte{}
 		s.Write(strconv.AppendInt(buf[:0], int64(line), 10))
 
+	case 'k':
+		name := c.fn.Name()
+		var path, pkg string
+		const pathSep = "/"
+		if i := strings.LastIndex(name, pathSep); i != -1 {
+			path, name = name[:i], name[i+len(pathSep):]
+		}
+		const pkgSep = "."
+		if i := strings.Index(name, pkgSep); i != -1 {
+			pkg, name = name[:i], name[i+len(pkgSep):]
+		}
+		if s.Flag('+') && path != "" {
+			pkg = path + "/" + pkg
```

Comment:
It should be possible to avoid this string concatenation. The original value of `name` already contains this data if sliced correctly. Rather than the `path` and `pkg` variables, keep track of `start` and `end` slice indices. Then I think it's as simple as `if s.Flag('+') { start = 0 }` before writing the string out.